### DISCUSSION TITLE
主笔记工作区笔记丢失且无法编辑该如何处理，有办法复原吗😭

### DISCUSSION THREAD

**Discussion by kcharlinan** [windingwind/zotero-better-notes#784] [Q&A]
我的笔记工作区在阅读文献时添加了很多批注，中间闪退过几次，但打开还存在。直到昨天最后一次闪退时，软件再打开我的笔记工作区变成了一片空白，下面的相关和标签还在，但笔记工作区的内容变成空白，且现在无法编辑，只能在单个文献中新建新的笔记，请问这种情况该如何处理呢，还有办法复原吗
![e7ee16f68e5569bee5dda2027e4d38f](https://github.com/windingwind/zotero-better-notes/assets/144091801/4768a8f8-49e5-4e66-8329-5bf08e1be8a0)

**Comment by windingwind** [comment]
请提供：
- 插件版本
- zotero版本
- 数据同步方式（官方/官方+webdav/其他）
- 在库中选中工作区笔记的截图

**Reply by windingwind** [reply]
请自查是否有此类情形，如果属于**将数据目录放在云同步服务中**的情形，爱莫能助。
https://www.zotero.org/support/kb/data_directory_in_cloud_storage_folder

**Reply by iamlgdm** [reply]
我的情况有点类似，两台电脑云同步，和楼主的zotero，插件版本一样，但是用坚果云同步，今天白天编辑好保存，晚上是保存好了但是无法编辑，想请教一下怎么回事呢？
![image](https://github.com/windingwind/zotero-better-notes/assets/158118731/0ae3b912-a437-4ace-9998-3967e66efbee)

**Reply by angkuLC** [reply]
大佬您好，我遇到了类似的问题，我的笔记只留下了一个标题。已尝试回滚到之前版本的本地备份，还是不行：
插件版本 1.0.4
zotero版本 7.0.0-beta.54+6b996d4f9 (64-bit)
数据同步方式（官方/官方+webdav/其他）官方
在库中选中工作区笔记的截图
![image](https://github.com/windingwind/zotero-better-notes/assets/61234473/b7a45359-3170-43b6-8d7e-89d087b3e42a)

调试日志：
`
1716490153559	addons.xpi	WARN	Checking C:\Program Files\Zotero\distribution\extensions for addons

[JavaScript Error: "NS_NOINTERFACE: " {file: "resource://gre/modules/ComponentUtils.jsm" line: 90}]
createInstance@resource://gre/modules/ComponentUtils.jsm:90:36


1716490154174	addons.xpi	WARN	Addon with ID undefined already installed, older version will be disabled

[JavaScript Error: "Error: Addon must include an id, version, and type" {file: "resource://gre/modules/addons/XPIProvider.jsm" line: 1710}]

[JavaScript Error: "Error: Addon must include an id, version, and type" {file: "resource://gre/modules/addons/XPIProvider.jsm" line: 1710}]

[JavaScript Error: "Passing a URI to Zotero.File.getContentsAsync() is deprecated -- use Zotero.HTTP.request() instead"]

[JavaScript Error: "uncaught exception: Object"]

[JavaScript Error: "uncaught exception: Object"]

[JavaScript Error: "uncaught exception: Object"]

[JavaScript Error: "uncaught exception: Object"]

[JavaScript Error: "uncaught exception: Object"]

[JavaScript Error: "Passing a URI to Zotero.File.getContentsAsync() is deprecated -- use Zotero.HTTP.request() instead"]

[JavaScript Error: "uncaught exception: Object"]

[JavaScript Error: "uncaught exception: Object"]

[JavaScript Error: "uncaught exception: Object"]

[JavaScript Error: "uncaught exception: Object"]

[JavaScript Error: "uncaught exception: Object"]

[JavaScript Error: "uncaught exception: Object" {file: "chrome://zotero/content/xpcom/file.js" line: 335}]

[JavaScript Error: "uncaught exception: Object"]

[JavaScript Error: "NotFoundError: Could not get children of file(C:\Users\angku\AppData\Roaming\Zotero\Zotero\Crash Reports\events) because it does not exist" {file: "resource://gre/modules/CrashManager.jsm" line: 682}]
_getUnprocessedEventsFiles/<@resource://gre/modules/CrashManager.jsm:682:12


[JavaScript Error: "uncaught exception: Object"]

[JavaScript Error: "uncaught exception: Object"]

[JavaScript Error: "TypeError: emptyListMessage is null" {file: "chrome://zotero/content/standalone/standalone.js" line: 679}]
ZoteroStandalone</this.updateAddonsPane/<@chrome://zotero/content/standalone/standalone.js:679:7


[JavaScript Error: "uncaught exception: Object"]

[JavaScript Error: "uncaught exception: Object"]

[JavaScript Error: "uncaught exception: Object"]

[JavaScript Error: "uncaught exception: Object"]

[JavaScript Error: "uncaught exception: Object"]

[JavaScript Error: "uncaught exception: Object"]

[JavaScript Error: "uncaught exception: Object"]

appName => Zotero, version => 7.0.0-beta.54+6b996d4f9, os => Windows 10.0 22631, locale => zh-CN, extensions => Translate for Zotero (1.1.0-7, extension), Chartero (2.3.3, extension), Zotero Better Notes (1.0.4, extension), Green Frog (0.19.01, extension), Awesome GPT (0.8.9, extension), Ethereal Style (4.4.6, extension)
`

谢谢大佬！

**Comment by kcharlinan** [comment]
·插件版本：1.0.4
·Zotero版本：6.0.30.0
·数据同步方式：（官方+webdav）
·在库中选中工作区笔记的截图：
![image](https://github.com/windingwind/zotero-better-notes/assets/144091801/b348e780-fdbf-4fde-bbc2-a7b9b517c004)
·数据同步截图
![image](https://github.com/windingwind/zotero-better-notes/assets/144091801/8ce0fd81-c44b-4ea7-a26b-c0a4c9e8c972)

**Reply by windingwind** [reply]
要恢复到先前某个时间点到数据，见https://www.zotero.org/support/zotero_data#restoring_your_zotero_data_from_a_backup

关于数据丢失的原因，可提供调试输出（menu - help - debug output - view output， 然后关闭工作区并重新打开，然后将debug output传到此处）。我可协助查看，但无法保证能够排除；也可在zotero forums发帖寻求帮助。

由于未有先前案例报告使用bn插件导致数据丢失，我倾向于推测是由于其他未知原因导致的闪退或数据丢失（两者之间未必存在必然联系）。

**Reply by kcharlinan** [reply]
[debug output.pdf](https://github.com/windingwind/zotero-better-notes/files/13327401/debug.output.pdf)
这一周尝试了几遍都不行，现在主笔记工作区可以编辑了，但是尝试恢复数据的方法还是不行，谢谢你的细致解答！这是之前的debug output文件

**Reply by windingwind** [reply]
抱歉该调试输出没有和本插件或数据丢失相关的线索，无法进一步排查您的问题。
数据恢复并非百分百能成功。
另外关于闪退，是否和https://github.com/windingwind/zotero-better-notes/issues/205 相同？编辑公式后输入中文产生闪退是平台层面的已知问题，目前zotero7计划升级基础平台，或可有所解决。可等待zotero7的正式版发布

**Reply by kcharlinan** [reply]
好吧，闪退问题和已给案例基本一致，等待软件修复中文相关问题。谢谢您的回答。

**Comment by Joseph1015** [comment]
想问一下主工作区笔记是怎么能够重新编辑的？我的插件版本和zotero版本同版主一致

**Reply by iamlgdm** [reply]
是指清空然后重写吗？这个直接创建独立笔记然后设置为工作区笔记也可以的。
